docs/src/main/paradox/server-class.md

```diff
@@ -0,0 +1,117 @@
+
+### The complete Route
+
+Below is the complete `Route` definition used in the sample application:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #all-routes }
+
+So far we have referred to `Route` without explaining what it is but now is the time to do so. Under the hood, Akka Http uses [Akka Streams](http://doc.akka.io/docs/akka/current/scala/stream/index.html). We don't have time to cover Akka Streams here, but if you are interested, you should take a look at the Hello World sample application for Streams. Since Akka Http is built on top of Akka Streams, it means that some concepts of Streams are available for us to use. In the case of `Route` you can think of it as a flow of in- and outbound data which is a perfect fit for Akka Streams. (Technically the `Route` type is `RequestContext ⇒ Future[RouteResult]` but there is no need to worry about what that means now.)
+
+## Http server
+
+To set up an Akka Http server we must first define some implicit values that will be used by the server:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #server-bootstrapping }
+
+What does the above mean and why do we need it?
+
+* `ActorSystem` : the context in which actors will run. What actors, you may wonder? Akka Streams uses actors under the hood, and the actor system defined in this `val` will be picked up and used by Streams.
+* `ActorMaterializer` : also Akka Streams related - it uses the materializer to allocate all the necessary resources it needs to run.
+
+With that defined we can move on to instantiate the server:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #http-server }
+
+We provide three parameters; `routes`, the hostname, and the port. That's it! When running this program, we will have an Akka Http server on our machine (localhost) on port 8080. Note that starting a server happens asynchronously and therefore a `Future` is returned by the `bindAndHandle` method.
+
+We should also add code for stopping the server. To do so we use the `StdIn.readLine()` method that will wait until RETURN is pressed on the keyboard. When that happens we `flatMap` the `Future` returned when we started the server to get to the `unbind()` method. Unbinding is also an asynchronous function and when the `Future` returned by `unbind()` is completes we make sure that the actor system is properly terminated.
```

Comment:
I'll let the jury decide what action to take here. I am open to both approaches.